guava/src/com/google/common/util/concurrent/EvictingBlockingQueue.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (C) 2010 The Guava Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.common.util.concurrent;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static com.google.common.base.Preconditions.checkNotNull;
+import java.io.Serializable;
+import java.util.concurrent.ArrayBlockingQueue;
+import java.util.concurrent.BlockingQueue;
+import java.util.concurrent.TimeUnit;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.errorprone.annotations.CanIgnoreReturnValue;
+
+/**
+ * A thread-safe implementation of the EvictingQueue data structure. It holds an ArrayBlockingQueue as the delegate
+ * which performers most of the BlockingQueue interface API. It overrides the methods that adds new objects to the
+ * data structure in order to implement the Evicting feature.
+ * 
+ * <p>An evicting blocking queue must be configured with a maximum size. Each time an element is added to a
+ * full queue, the queue automatically removes its head element. This is different from conventional
+ * bounded queues, which either block or reject new elements when full.
+ * 
+ * <p>This class does not accept null elements.
+ * 
+ * @author Noam Greenshtain
+
+ */
+
+public final class EvictingBlockingQueue<E> extends ForwardingBlockingQueue<E> implements Serializable{
+	
+	private static final long serialVersionUID = -4305601088881265868L;
```

Comment:
https://google.github.io/styleguide/javaguide.html#s5.2.4-constant-names

Constants should be uppercase snake?

Reply:
The property was removed.